as-ap-opticscorr/as_ap_opticscorr/chrom/main.py

```diff
@@ -427,78 +429,71 @@ def _callback_estimate_chrom(self, pvname, value, **kws):
-               else 0)
         self._status = _siriuspy.util.update_bit(
-            v=self._status, bit_pos=1, bit_val=val)
+            v=self._status, bit_pos=1,
+            bit_val=any(s != _PSConst.PwrStateSts.On
+                        for s in self._sfam_check_pwrstate_sts))
         self.driver.setParam('Status-Mon', self._status)
         self.driver.updatePVs()
 
     def _callback_sfam_opmode_sts(self, pvname, value, **kws):
-        ps = pvname.split(self._PREFIX_VACA)[1]
-        self.driver.setParam('Log-Mon', 'WARN:'+ps+' changed.')
+        self.driver.setParam('Log-Mon', 'WARN:'+pvname+' changed.')
         self.driver.updatePVs()
 
-        fam = ps.split(':')[1].split('-')[1]
-        fam_index = self._SFAMS.index(fam)
+        fam_index = self._SFAMS.index(_SiriusPVName(pvname).dev)
         self._sfam_check_opmode_sts[fam_index] = value
 
         # Change the third bit of correction status
-        opmode = self._sync_corr
-        val = (1 if any(s != opmode for s in self._sfam_check_opmode_sts)
-               else 0)
+        opmode = _PSConst.States.SlowRefSync if self._sync_corr \
+            else _PSConst.States.SlowRef
         self._status = _siriuspy.util.update_bit(
-            v=self._status, bit_pos=2, bit_val=val)
+            v=self._status, bit_pos=2,
+            bit_val=any(s != opmode for s in self._sfam_check_opmode_sts))
         self.driver.setParam('Status-Mon', self._status)
         self.driver.updatePVs()
 
     def _callback_sfam_ctrlmode_mon(self,  pvname, value, **kws):
-        ps = pvname.split(self._PREFIX_VACA)[1]
-        if value == 1:
-            self.driver.setParam('Log-Mon', 'WARN:'+ps+' is Local.')
+        if value != _PSConst.Interface.Remote:
+            self.driver.setParam('Log-Mon', 'WARN:'+pvname+' is not Remote.')
             self.driver.updatePVs()
 
-        fam = ps.split(':')[1].split('-')[1]
-        fam_index = self._SFAMS.index(fam)
+        fam_index = self._SFAMS.index(_SiriusPVName(pvname).dev)
         self._sfam_check_ctrlmode_mon[fam_index] = value
 
         # Change the fourth bit of correction status
-        val = (1 if any(s == 1 for s in self._sfam_check_ctrlmode_mon)
-               else 0)
         self._status = _siriuspy.util.update_bit(
-            v=self._status, bit_pos=3, bit_val=val)
+            v=self._status, bit_pos=3,
+            bit_val=any(s != _PSConst.Interface.Remote
+                        for s in self._sfam_check_ctrlmode_mon))
         self.driver.setParam('Status-Mon', self._status)
         self.driver.updatePVs()
 
     def _callback_timing_state(self, pvname, value, **kws):
         if 'Sexts:State' in pvname:
             self._timing_check_config[0] = value  # Enbl
         elif 'Sexts:EVGParam' in pvname:
-            self._timing_check_config[1] = (1 if value == 1 else 0)  # Chroms
+            self._timing_check_config[1] = (value == 1)  # Chroms
```

Comment:
Please, use information from the `siriuspy.csdevice.timesys` module to make this comparison. Do not use a hard coded value, because this index can change.